eth/client.go

```diff
@@ -1101,6 +1089,8 @@ func (c *client) CheckTx(tx *types.Transaction) error {
 
 	for {
 		select {
+		case <-time.After(c.checkTxTimeout):
```

Comment:
I believe you actually need to create the `time.After` timer only once, outside the `for` loop, otherwise you will reset the timer for every receipt that you get even if not related to the tx you're checking.

Reply:
Updated in a354f7148d50b275912ac379f405d2f7dd9124ca